File: heart/src/main/java/com/vantagecircle/heartrate/fragment/BaseFragment.java
package com.vantagecircle.heartrate.fragment;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

/**
 * Created by bapidas on 08/11/17.
 */

public abstract class BaseFragment extends Fragment {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setFragmentComponent();
    }

    protected abstract void setFragmentComponent();

    protected abstract void init();
}

File: app/src/main/java/com/bapidas/heartrate/ui/base/BaseFragment.kt
package com.bapidas.heartrate.ui.base

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment

/**
 * Created by bapidas on 08/11/17.
 */
abstract class BaseFragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context?.let { setFragmentComponent(it) }
    }

    protected abstract fun setFragmentComponent(context: Context)

    protected abstract fun init()
}